### PR TITLE
Feature/plcn 201 add description and name to deal entity

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**

### CODE REVIEW

File: Pelican.Domain/Entities/AccountManager.cs [hunk 1]
```diff
@@ -3,33 +3,71 @@
...
+		}
+	}
+	private string _lastName = string.Empty;
+	public string LastName
+	{
+		get => _lastName;
+		set
+		{
+			_lastName = value.Length > StringLengths.Name ? value.Substring(0, StringLengths.Name - 3) + ("...") : value;
+		}
```
Comment (Nicolai-Bak) [inline Pelican.Domain/Entities/AccountManager.cs:26]: Would probably have done something like this as it is more readable to me, but i don't think it is necessary to change now. 

``` C#
set => _lastName = value.Length > StringLengths.Name 
	? $"{value[..(StringLengths.Name - 3)]}..." 
	: value;
```

File: Pelican.Domain/Entities/AccountManager.cs [hunk 2]
```diff
@@ -3,33 +3,71 @@
...
-
-	public string? PhoneNumber { get; set; }
-
-	public string? PictureUrl { get; set; }
-
-	public string? LinkedInUrl { get; set; }
-
-
+	private string _firstName = string.Empty;
+	public string FirstName
```
Comment (Nicolai-Bak) [inline Pelican.Domain/Entities/AccountManager.cs:11]: In classes i would prefer to order things by types, eg:
- Private fields
- Constructors
- Properties

File: Pelican.Domain/Entities/AccountManager.cs [hunk 3]
```diff
@@ -3,33 +3,71 @@
...
+			_phoneNumber = value!.Length > StringLengths.PhoneNumber ? value.Substring(0, StringLengths.PhoneNumber - 3) + ("...") : value;
+		}
+	}
+	private string? _pictureUrl;
+	public string? PictureUrl
+	{
+		get => _pictureUrl;
+		set
+		{
+			_pictureUrl = value!.Length > StringLengths.Url ? value.Substring(0, StringLengths.Url - 3) + ("...") : value;
```
Comment (Nicolai-Bak) [inline Pelican.Domain/Entities/AccountManager.cs:52]: dont think this should be shortened

File: Pelican.Domain/Entities/Client.cs [hunk 4]
```diff
@@ -4,13 +4,37 @@
...
+	{
+		get => _name;
+		set
+		{
+			_name = value.Length > StringLengths.Name ? value.Substring(0, StringLengths.Name - 3) + ("...") : value;
+		}
+	}
 
-	public string HubSpotId { get; set; }
+	public string HubSpotId = string.Empty;
```
Comment (Nicolai-Bak) [inline Pelican.Domain/Entities/Client.cs:17]: missing get and set

**Review by Nicolai-Bak — COMMENTED** [review]
Commen:
- For entity tests: I would prefer to use a fixed string instead of using bogus as it removed logik from tests and makes it easier to see what the intention is.
- For entities I think it should be reconsidered when it is valid to cut a string

File: Pelican.Infrastructure.HubSpot/Extensions/RestRequestExtensions.cs [hunk 5]
```diff
@@ -14,6 +14,7 @@ internal static class RestRequestExtensions
 			.AddQueryParameter("properties", "notes_last_contacted", false)
 			.AddQueryParameter("properties", "dealstage", false)
 			.AddQueryParameter("properties", "dealname", false)
+			.AddQueryParameter("properties", "description", false)
 			.AddQueryParameter("properties", "amount", false);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Infrastructure.HubSpot/Extensions/RestRequestExtensions.cs:18]: Has been removed

File: Pelican.Infrastructure.HubSpot/Mapping/Clients/CompanyResponseToClient.cs [hunk 6]
```diff
@@ -15,10 +15,12 @@ internal static Client ToClient(this CompanyResponse response)
 
 		Client result = new(Guid.NewGuid())
 		{
-			Name = response.Properties.Name,
 			HubSpotId = response.Properties.HubSpotObjectId,
-			OfficeLocation = response.Properties.City,
 		};
+		result.Name = response.Properties.Name;
+		result.Website = response.Properties.Domain;
+		result.OfficeLocation = response.Properties.City;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Infrastructure.HubSpot/Mapping/Clients/CompanyResponseToClient.cs:22]: Of course, this is left over from when the string length check was here.

File: Pelican.Infrastructure.HubSpot/Mapping/AccountManagers/OwnerResponseToAccountManager.cs [hunk 7]
```diff
@@ -15,14 +15,15 @@ internal static AccountManager ToAccountManager(this OwnerResponse response)
...
 		{
 			HubSpotId = response.Id,
-			Email = response.Email,
-			FirstName = response.Firstname,
-			LastName = response.Lastname,
 			HubSpotUserId = response.UserId,
 		};
+		result.Email = response.Email;
+		result.FirstName = response.Firstname;
+		result.LastName = response.Lastname;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Infrastructure.HubSpot/Mapping/AccountManagers/OwnerResponseToAccountManager.cs:25]: Of course, this is left over from when the string length check was here.

File: Pelican.Domain/Entities/Supplier.cs [hunk 8]
```diff
@@ -6,21 +6,73 @@ public class Supplier : Entity, ITimeTracked
...
-
+	private string _refreshToken = string.Empty;
+	public string RefreshToken
+	{
+		get => _refreshToken;
+		set
+		{
+			_refreshToken = value!.Length > StringLengths.Token ? value.Substring(0, StringLengths.Token - 3) + ("...") : value;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/Supplier.cs:17]: Makes sense

File: Pelican.Domain/Entities/Supplier.cs [hunk 9]
```diff
@@ -6,21 +6,73 @@ public class Supplier : Entity, ITimeTracked
...
+
+	private string? _pictureUrl;
+	public string? PictureUrl
+	{
+		get => _pictureUrl;
+		set
+		{
+			_pictureUrl = value!.Length > StringLengths.Url ? value.Substring(0, StringLengths.Url - 3) + ("...") : value;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/Supplier.cs:37]: Ít no longer is shortened

File: Pelican.Domain.Test/Entities/LocationUnitTest.cs [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+		// Arrange
+		Faker faker = new();
+		string propertyValue = faker.Lorem.Letter(StringLengths.OfficeLocation * 2);
+
+		// Act
+		_uut.CityName = propertyValue;
+		// Assert
+		Assert.Equal(StringLengths.OfficeLocation, _uut.CityName!.Length);
+		Assert.Equal("...", _uut.CityName.Substring(StringLengths.OfficeLocation - 3));
+		Assert.Equal(propertyValue.Substring(0, StringLengths.OfficeLocation - 3), _uut.CityName.Substring(0, StringLengths.OfficeLocation - 3));
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain.Test/Entities/LocationUnitTest.cs:41]: Fixed it as we talked about

File: Pelican.Domain/Entities/AccountManager.cs (hunk 1 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/AccountManager.cs:26]: I think that is a good idea as well

File: Pelican.Domain/Entities/AccountManager.cs (hunk 2 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/AccountManager.cs:11]: Done

File: Pelican.Domain/Entities/AccountManager.cs (hunk 3 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/AccountManager.cs:52]: It no longer is

File: Pelican.Domain/Entities/Client.cs (hunk 4 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Entities/Client.cs:17]: Whoops, it is back

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**